platform/openide.loaders/src/org/openide/loaders/TemplateWizard.java

```diff
@@ -47,6 +48,13 @@
 * @see TemplateRegistration
 */
 public class TemplateWizard extends WizardDescriptor {
+    /** Set to <CODE>File</CODE> to use for deferred creation of target folder.
+     * If <CODE>TemplateWizard.getTargetFolder()</CODE> is called and the
+     * the target folder has not been set then this property value is
+     * used to create the directory and DataFolder. This allows deferral
+     * of creation of DataFolder until and <em>if</em> it is needed.
+     */
```

Comment:
This would be an API change and as such it would need additional attributes of an API change. See the [wiki page](http://wiki.netbeans.org/APIDevelopment) for overall description. Here is the enumeration of what I am missing the most:
- incrementing module version, using `@since` tag
- modifying `apichanges.xml`
- testing the change